htlcswitch/hop/fuzz_test.go

```diff
@@ -122,15 +122,26 @@ func fuzzPayload(f *testing.F, finalPayload bool) {
 		var b bytes.Buffer
 		hop, nextChanID := hopFromPayload(payload1)
 		err = hop.PackHopPayload(&b, nextChanID, finalPayload)
-		if errors.Is(err, route.ErrAMPMissingMPP) {
-			// PackHopPayload refuses to encode an AMP record
-			// without an MPP record. However, NewPayloadFromReader
-			// does allow decoding an AMP record without an MPP
-			// record, since validation is done at a later stage. Do
-			// not report a bug for this case.
+		switch {
+		// PackHopPayload refuses to encode an AMP record
+		// without an MPP record. However, NewPayloadFromReader
+		// does allow decoding an AMP record without an MPP
+		// record, since validation is done at a later stage. Do
+		// not report a bug for this case.
+		case errors.Is(err, route.ErrAMPMissingMPP):
 			return
+
+		// PackHopPayload will not encode regular payloads or final
+		// hops in blinded routes that do not have an amount or expiry
+		// TLV set. However, NewPayloadFromReader will allow creation
+		// of payloads where these TLVs are present, but they have
+		// zero values because validation is done at a later stage.
```

Comment:
Q: Do we validate the zero values also when forwarding an HTLC or what do you mean exactly by:

```validation is done at a later stage ?```

Reply:
Just referring to the validation that we do when adding the HTLC to our channel (which disallows zero values). Copied the phrasing from the comment for AMP above.

Reply:
Cool, I haven't looked into the Forwarding Blinded Payments PR but will we allow Onions which have the TLV value set but the for example the value is set to 0 pass through ?

Reply:
>  Onions which have the TLV value set but the for example the value is set to 0 pass through ?

By the time we get to the outgoing link, we've calculated the amount that we _actually_ need to forward from the encrypted data.